File: Codes/Python/SEM 1/Assignment_Class/time.py
#TIME CLASS

class Time():

    def __init__(self,hour,minute,second):
        self.__hour:int = hour
        self.__minute:int = minute
        self.__second:int = second
    
    def getHour(self) -> int:
        print(self.__hour)

    def getMinute(self) -> int:
        print(self.__minute)
    
    def getSecond(self) -> int:
        print(self.__second)
    
    def setHour(self,hour:int):
        self.__hour = hour
        print(self.__hour)
    
    def setMinute(self,minute:int):
        self.__minute = minute
        print(self.__minute)
    
    def setSecond(self,second:int):
        self.__second = second
        print(self.__second)
    
    def setTime(self,hour:int,minute:int,second:int):
        self.__hour = hour
        self.__minute = minute
        self.__second = second
    
    def toString(self) -> str:
        zero = "0"
        time = [str(self.__hour),str(self.__minute),str(self.__second)]
        for i in range(len(time)):
            if len(time[i]) == 1:
                time[i] = zero + time[i]

        print(time[0] + ':' + time[1] + ':' + time[2])

    def nextSecond(self):
        self.__second += 1
        if self.__second >= 60:
            self.__second -= 60
            self.__minute += 1
            if self.__minute >= 60:
                self.__minute -= 60
                self.__hour += 1
        
    
    def previousSecond(self):
        self.__second -= 1
        if self.__second < 0:
            self.__second = 60 + self.__second
            self.__minute -= 1
            if self.__minute < 0:
                self.__minute = 60 + self.__minute
                self.__hour -= 1

#Tests
                    
time1 = Time(4,40,40)
time1.getHour()
time1.getMinute()
time1.getSecond()
time1.setHour(5)
time1.setMinute(50)
time1.setSecond(50)
time1.setTime(6,59,59)
time1.toString()
time1.nextSecond()
time1.toString()
time1.previousSecond()
time1.toString()

File: Codes/JAVA/classexercise/timeclass/time.java
package timeclass;

public class time
{
    private int hour;
    private int minute;
    private int second;

    public time(int hour, int minute, int second)
    {
        if (hour >= 0 && hour<= 23)
        {
            if (minute >= 0 && minute <= 59)
            {
                if (second >= 0 && second <= 59)
                {
                    this.hour = hour;
                    this.minute = minute;
                    this.second = second;
                }
            }
        }
    }

    public int getHour()
    {
        return hour;
    }

    public int getMinute()
    {
        return minute;
    }

    public int getSecond()
    {
        return second;
    }

    public void setHour(int hour)
    {
        if (hour >= 0 && hour <= 23)
        {
            this.hour = hour;
        }
    }

    public void setMinute(int minute)
    {
        if (minute >= 0 && minute <= 59)
        {
            this.minute = minute;
        }
    }

    public void setSecond(int second)
    {
        if (second >= 0 && second <= 23)
        {
            this.second = second;
        }
    }

    public void setTime(int hour,int minute,int second)
    {
        if (hour >= 0 && hour<= 23)
        {
            if (minute >= 0 && minute <= 59)
            {
                if (second >= 0 && second <= 59)
                {
                    this.hour = hour;
                    this.minute = minute;
                    this.second = second;
                }
            }
        }
    }

    public String toString()
    {
        if (hour < 10)
        {
            if (minute < 10)
            {
                if (second < 10)
                {
                    return "0" + hour + ":0" + minute + ":0" + second;
                }

                else
                {
                    return "0" + hour + ":0" + minute + ":" + second;
                }
            }

            else
            {
                if (second < 10)
                {
                    return "0" + hour + ":" + minute + ":0" + second;
                }

                else
                {
                    return "0" + hour + ":" + minute + ":" + second;
                }
            }
        }

        else
        {
            if (minute < 10)
            {
                if (second < 10)
                {
                    return hour + ":0" + minute + ":0" + second;
                }

                else
                {
                    return hour + ":0" + minute + ":" + second;
                }
            }

            else
            {
                if (second < 10)
                {
                    return hour + ":" + minute + ":0" + second;
                }

                else
                {
                    return hour + ":" + minute + ":" + second;
                }
            }
        }
    }

    public time nextSecond()
    {
        second += 1;
        if (second > 59)
        {
            second = 0;
            minute += 1;
            if(minute > 59)
            {
                minute = 0;
                hour += 1;
                if (hour > 23)
                {
                    hour = 0;
                }
            }
        }

        return this;
    }

    public time previousSecond()
    {
        second -= 1;
        if(second == -1)
        {
            second = 59;
            minute -= 1;
            if (minute == -1)
            {
                minute = 59;
                hour -= 1;
                if (hour == -1)
                {
                    hour = 23;
                }
            }
        }

        return this;
    }
};